src/neuroglancer/mesh/backend.ts

```diff
@@ -358,24 +365,33 @@ export class MeshSource extends ChunkSource {
     fragmentSource.meshSource = this;
   }
 
-  getChunk(objectId: Uint64) {
+  getChunk(objectId: Uint64, verifyFragments?: boolean|undefined) {
     const key = getObjectKey(objectId);
     let chunk = <ManifestChunk>this.chunks.get(key);
     if (chunk === undefined) {
+      if (verifyFragments === undefined) verifyFragments = true;
       chunk = this.getNewChunk_(ManifestChunk);
-      chunk.initializeManifestChunk(key, objectId);
+      chunk.initializeManifestChunk(key, objectId, verifyFragments);
       this.addChunk(chunk);
     }
     return chunk;
   }
 
-  getFragmentChunk(manifestChunk: ManifestChunk, fragmentId: FragmentId) {
-    let key = `${manifestChunk.key}/${fragmentId}`;
+  getFragmentChunk(manifestChunk: ManifestChunk, fragmentId: FragmentId, verifyFragment?: boolean|undefined) {
+    let key = fragmentId;
+    if (fragmentId.charAt(0) === '~') {
+      // extract segment ID from fragment ID
+      // use it as key, the rest is information for reading the fragment
+      let parts = fragmentId.substr(1).split(/:(.+)/);
+      key = parts[0];
```

Comment:
Looking at part of the response for https://minniev1.microns-daf.com/meshing/api/v1/table/minnie3_v0/manifest/864691135611454232:0?verify=True
```js
 [
    "~2/317949958-0.shard:378515:218", 
    "~2/267612164-0.shard:498888:241", 
    "~2/267612164-0.shard:503532:213", 
]
```
it seems this code would assign `2/267612164-0.shard` as key for both fragments?

Reply:
no the whole string is used as fragment ID so there's no conflict